gonum fails to build when ints are 32 bits

Running tests with GOARCH=386 results in:

```
# gonum.org/v1/gonum/graph/encoding/digraph6 [gonum.org/v1/gonum/graph/encoding/digraph6.test]
./digraph6.go:65:9: constant 68719476736 overflows int
FAIL	gonum.org/v1/gonum/graph/encoding/digraph6 [build failed]
```

And same problem in graph6.

### What version of Go and Gonum are you using?

Tested with 0.8.1 and HEAD (1a536f9a771e1c098630abc9a8bf53b7e5e1994f) with go1.15 on linux

### Does this issue reproduce with the current master?

Yes

This simple patch fixes it but doesn't seem very elegant

```
diff --git a/graph/encoding/digraph6/digraph6.go b/graph/encoding/digraph6/digraph6.go
index 5b53136f..f172e41e 100644
--- a/graph/encoding/digraph6/digraph6.go
+++ b/graph/encoding/digraph6/digraph6.go
@@ -62,7 +62,7 @@ func Encode(g graph.Graph) Graph {
 		buf.WriteByte(byte(n) + 63)
 	case n < 258048:
 		buf.Write([]byte{126, byte(n>>12) + 63, byte(n>>6) + 63, byte(n) + 63})
-	case n < 68719476736:
+	case int64(n) < 68719476736:
 		buf.Write([]byte{126, 126, byte(n>>30) + 63, byte(n>>24) + 63, byte(n>>18) + 63, byte(n>>12) + 63, byte(n>>6) + 63, byte(n) + 63})
 	default:
 		panic("digraph6: too large")
diff --git a/graph/encoding/graph6/graph6.go b/graph/encoding/graph6/graph6.go
index 90c84117..18153a83 100644
--- a/graph/encoding/graph6/graph6.go
+++ b/graph/encoding/graph6/graph6.go
@@ -63,7 +63,7 @@ func Encode(g graph.Graph) Graph {
 		buf.WriteByte(byte(n) + 63)
 	case n < 258048:
 		buf.Write([]byte{126, byte(n>>12) + 63, byte(n>>6) + 63, byte(n) + 63})
-	case n < 68719476736:
+	case int64(n) < 68719476736:
 		buf.Write([]byte{126, 126, byte(n>>30) + 63, byte(n>>24) + 63, byte(n>>18) + 63, byte(n>>12) + 63, byte(n>>6) + 63, byte(n) + 63})
 	default:
 		panic("graph6: too large")
```

gonum.org/v1/gonum/graph/layout also has test failings but this is not a build problem this time and may just be the test:

```
--- FAIL: TestIsomapR2 (1.13s)
    layout_test.go:81: image mismatch for testdata/sheet_isomap.png
    layout_test.go:95: image bounds union:(0,0)-(378,378) diff bounds intersection:(0,0)-(378,378)
    layout_test.go:103: IMAGE:iVBORw0KG[...]
    isomap_test.go:187: got node positions: map[int64]r2.Vec{0:r2.Vec{X:-0.0628653502565152, Y:-2.0451113940755437}, 1:r2.Vec{X:-0.8967133438226523, Y:-0.8432286945344888}, 2:r2.Vec{X:-2.0451113940755445, Y:0.06286535025651527}, 3:r2.Vec{X:0.8432286945344889, Y:-0.8967133438226514}, 4:r2.Vec{X:-2.2110514424397126e-16, Y:-5.755313712676207e-17}, 5:r2.Vec{X:-0.8432286945344893, Y:0.8967133438226519}, 6:r2.Vec{X:2.0451113940755445, Y:-0.06286535025651524}, 7:r2.Vec{X:0.8967133438226521, Y:0.8432286945344887}, 8:r2.Vec{X:0.06286535025651516, Y:2.0451113940755437}}
    layout_test.go:81: image mismatch for testdata/tube_isomap.png
    layout_test.go:95: image bounds union:(0,0)-(378,378) diff bounds intersection:(0,0)-(378,378)
    layout_test.go:103: IMAGE:iVBORw0KG[...]
    isomap_test.go:187: got node positions: map[int64]r2.Vec{0:r2.Vec{X:-1.2909944487358054, Y:-0.8201535052616763}, 1:r2.Vec{X:-1.290994448735805, Y:0.9479814504843308}, 2:r2.Vec{X:-1.2909944487358045, Y:-0.1278279452226542}, 3:r2.Vec{X:-2.4765360345773215e-16, Y:-0.6628012871097505}, 4:r2.Vec{X:-1.4535345623533508e-16, Y:0.7661045420231556}, 5:r2.Vec{X:3.283287848946293e-16, Y:-0.10330325491340546}, 6:r2.Vec{X:1.2909944487358047, Y:-0.820153505261676}, 7:r2.Vec{X:1.2909944487358047, Y:0.9479814504843298}, 8:r2.Vec{X:1.2909944487358056, Y:-0.12782794522265395}}
FAIL
exit status 1
```

--- a/graph/encoding/digraph6/digraph6.go
+++ b/graph/encoding/digraph6/digraph6.go
@@ -57,7 +57,12 @@ func Encode(g graph.Graph) Graph {
 
 	var buf strings.Builder
 	buf.WriteByte('&')
-	switch {
+	// digraph6 specifies graphs of order up to 68719476735
+	// which overflows int on 32 bit architectures. We know
+	// that on those machines n will not be this large, since
+	// it came from a length, but explicitly convert to 64
+	// bits to allow the package to build on those architectures.
+	switch n := int64(n); {
 	case n < 63:
 		buf.WriteByte(byte(n) + 63)
 	case n < 258048:

--- a/graph/encoding/digraph6/digraph6.go
+++ b/graph/encoding/digraph6/digraph6.go
@@ -57,11 +57,11 @@ func Encode(g graph.Graph) Graph {
 
 	var buf strings.Builder
 	buf.WriteByte('&')
-	// digraph6 specifies graphs of order up to 68719476735
-	// which overflows int on 32 bit architectures. We know
-	// that on those machines n will not be this large, since
-	// it came from a length, but explicitly convert to 64
-	// bits to allow the package to build on those architectures.
+	// digraph6 specifies graphs of order up to 2^36-1 which
+	// overflows int on 32-bit architectures. We know that on
+	// those machines n will not be this large, since it came
+	// from a length, but explicitly convert to 64 bits to
+	// allow the package to build on those architectures.
 	switch n := int64(n); {
 	case n < 63:
 		buf.WriteByte(byte(n) + 63)

--- a/graph/encoding/graph6/graph6.go
+++ b/graph/encoding/graph6/graph6.go
@@ -58,7 +58,12 @@ func Encode(g graph.Graph) Graph {
 	}
 
 	var buf strings.Builder
-	switch {
+	// graph6 specifies graphs of order up to 68719476735
+	// which overflows int on 32 bit architectures. We know
+	// that on those machines n will not be this large, since
+	// it came from a length, but explicitly convert to 64
+	// bits to allow the package to build on those architectures.
+	switch n := int64(n); {
 	case n < 63:
 		buf.WriteByte(byte(n) + 63)
 	case n < 258048:

--- a/graph/encoding/graph6/graph6.go
+++ b/graph/encoding/graph6/graph6.go
@@ -58,11 +58,11 @@ func Encode(g graph.Graph) Graph {
 	}
 
 	var buf strings.Builder
-	// graph6 specifies graphs of order up to 68719476735
-	// which overflows int on 32 bit architectures. We know
-	// that on those machines n will not be this large, since
-	// it came from a length, but explicitly convert to 64
-	// bits to allow the package to build on those architectures.
+	// graph6 specifies graphs of order up to 2^36-1 which
+	// overflows int on 32-bit architectures. We know that on
+	// those machines n will not be this large, since it came
+	// from a length, but explicitly convert to 64 bits to
+	// allow the package to build on those architectures.
 	switch n := int64(n); {
 	case n < 63:
 		buf.WriteByte(byte(n) + 63)

